pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -0,0 +1,55 @@
+package clusterauthorizer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/form3tech-oss/jwt-go"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclaim"
+	"github.com/Azure/ARO-RP/pkg/util/refreshable"
+)
+
+type Credentials struct {
+	ClientID     []byte
+	ClientSecret []byte
+	TenantID     []byte
+}
+
+//NewAuthorizer returns a new refreshable authorizer based on an auth token (see GetToken in /pkg/util/aad)
+func NewAuthorizer(token *adal.ServicePrincipalToken) (refreshable.Authorizer, error) {
```

Comment:
I changed it to NewAzRefreshableAuthorizer, so that covers 2nd and 3rd (I assume Az is enough to distinguish it from the others, would AzARM be better?)

I don't quite get 1st, but I think it will be void if I follow the refactoring suggestions re: wrapping it all into a function which just returns a working client instead of having all the steps in the reconciler.